packages/components/src/flex/index.js

```diff
@@ -33,32 +36,48 @@ export { default as FlexItem } from './item';
  * @param {Props} props
  * @param {import('react').Ref<HTMLDivElement>} ref
  */
-function FlexComponent(
-	{
+function FlexComponent( props, ref ) {
+	const direction = useDirection( props );
+	const {
 		align = 'center',
 		className,
 		gap = 2,
 		justify = 'space-between',
-		isReversed = false,
-		...props
-	},
-	ref
-) {
+		...otherProps
+	} = props;
```

Comment:
To fix `direction` bleeding though I think we may want to just destructure props earlier. The problem is that right now `direction` and `isReversed` will continue to live on `otherProps` so they'll get passed down.

```js
function FlexComponent( {
    align = 'center',
    className,
    gap = 2,
    justify = 'space-between',
    isReversed,
    direction: directionProp,
    ...otherProps
  },
  ref
) {
  const direction = useDirection( { isReversed, direction: directionProp } );

  // ...
}

Reply:
You know, I thought about that and will try it out here in a second.  But I expected that if we are returning the element like `<Thing ...otherProps direction={ direction } />` that if `otherProps` contained a 'direction' prop, that it would get overridden by the latter declaration?

Reply:
Just tried it in 	a64d6fd, it seems to have the same issue.. just moved to the end of the list 😆  :
![Screen Shot 2021-03-26 at 3 03 49 PM](https://user-images.githubusercontent.com/28742426/112681191-baedd080-8e44-11eb-9561-80e7f7433bcd.png)

Reply:
Ah! Aha, I was wrong about the reason 🤦‍♀️ 

Emotion will forward all props that are valid HTML attributes: https://emotion.sh/docs/styled#customizing-prop-forwarding

We can disable it following the instructions there for the `direction` prop.

Reply:
Oh thanks!  Thats definitely it 😁  54e68f5 seems to fix that issue.